pkg/docker/config/config.go

```diff
@@ -204,7 +208,17 @@ func GetAllCredentials(sys *types.SystemContext) (map[string]types.DockerAuthCon
 // The homeDir parameter should always be homedir.Get(), and is only intended to be overridden
 // by tests.
 func getAuthFilePaths(sys *types.SystemContext, homeDir string) []authPath {
+	runningInSystemd := os.Getenv("INVOCATION_ID") != ""
+	runningAsRoot := os.Getuid() == 0
+	runningSystemdPrivileged := runningInSystemd && runningAsRoot
+
 	paths := []authPath{}
+
+	// If we're in systemd, prefer the global auth path first.
+	if runningSystemdPrivileged {
```

Comment:
The explicit caller’s configuration from `types.SystemContext` should be preferred over any built-in defaults.

Reply:
It looks like you partially addressed this with https://github.com/containers/image/pull/1751 - I built on that, though this is getting messier.

Reply:
Looking back at #1751, that duplication really wasn’t a good idea. #1772 should help.